src/main/java/net/onenandone/fralax/parser/VtdXmlParserContext.java

```diff
@@ -224,54 +208,46 @@ public String asFormattedString() {
-                child = child + childChild;
-            }
-            child = child + "</" + navigation.toNormalizedString(curIndex) + ">";
-            children.add(child);
-            children.addAll(newChildren.get(1));
+            traverse(rootDepth, startDepth, childrenAndSiblings.children);
         } else {
-            children.add(navigation.getXPathStringVal());
+            childrenAndSiblings.children.add(navigation.getXPathStringVal());
         }
 
         //After traversing all children nodes we now go back to our parent element and traverse all our siblings.
         navigation.recoverNode(parentIndex);
         //Assignment so the next sibling traversal uses correct depth to determine if we should search for more children.
-        startDepth--;
+        final int newStartDepth = startDepth - 1;
         //Traversing siblings, uses rootDepth in the check so we don't keep on checking siblings of the node we start our search from.
         if (navigation.toElement(VTDNav.NEXT_SIBLING) && rootDepth < navigation.getCurrentDepth()) {
-            int curIndex = navigation.getCurrentIndex();
-            String sibling = "<";
-            sibling = sibling + navigation.toNormalizedString(curIndex);
-            for (String attribute : evaluateAttributes()) {
-                sibling = sibling + " " + attribute;
-            }
-            sibling = sibling + ">";
-            List<List<String>> newChildren = evaluateChildrenAndSiblings(rootDepth, curIndex, startDepth + 1);
-            for (String childChild : newChildren.get(0)) {
-                sibling = sibling + childChild;
-            }
-            sibling = sibling + "</" + navigation.toNormalizedString(curIndex) + ">";
-            siblings.add(sibling);
-            siblings.addAll(newChildren.get(1));
+            traverse(rootDepth, newStartDepth, childrenAndSiblings.siblings);
         }
-        return result;
+        return childrenAndSiblings;
+    }
+
+    private void traverse(final int rootDepth, final int startDepth, final List<String> elements) throws NavException {
+        int curIndex = navigation.getCurrentIndex();
+        String child = "<";
+        child = child + navigation.toNormalizedString(curIndex);
+        for (final String attribute : evaluateAttributes()) {
+            child = child + " " + attribute;
+        }
+        child = child + ">";
+        final ChildrenAndSiblings childrenAndSiblings = evaluateChildrenAndSiblings(rootDepth, curIndex, startDepth + 1);
+        for (final String childChild : childrenAndSiblings.children) {
+            child = child + childChild;
+        }
+        child = child + "</" + navigation.toNormalizedString(curIndex) + ">";
+        elements.add(child);
+        elements.addAll(childrenAndSiblings.siblings);
+    }
+
+    private static class ChildrenAndSiblings {
+        private final List<String> children = new ArrayList<>();
+        private final List<String> siblings = new ArrayList<>();
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://www.codacy.com/app/sysdev/fralax/file/2402105292/issues/source?bid=3227012&fileBranchId=3232616#l250)